build.gradle

```diff
@@ -1,21 +1,16 @@
 import com.github.jengelman.gradle.plugins.shadow.ShadowPlugin
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
 
 plugins {
     id 'groovy'
     id 'project-report'
     id 'idea'
     id 'java-gradle-plugin'
-    // id 'signing'
     id 'com.gradle.plugin-publish' version '1.1.0'
     id 'org.ajoberstar.git-publish' version '4.1.1'
     id 'com.github.node-gradle.node' version '3.5.1'
 }
 
-// Remove the gradleApi so it isn't merged into the jar file.
-configurations.named(JavaPlugin.API_CONFIGURATION_NAME) {
-    dependencies.remove(project.dependencies.gradleApi())
-}
```

Comment:
`compileOnly` should be enough

```sh
./gradlew assemble

diffuse diff --jar shadow-8.1.0-SNAPSHOT-before.jar shadow-8.1.0-SNAPSHOT-after.jar

OLD: shadow-8.1.0-SNAPSHOT-before.jar
NEW: shadow-8.1.0-SNAPSHOT-after.jar

 JAR   │ old       │ new       │ diff 
───────┼───────────┼───────────┼──────
 class │ 700.9 KiB │ 700.9 KiB │  0 B 
 other │  10.6 KiB │  10.6 KiB │  0 B 
───────┼───────────┼───────────┼──────
 total │ 711.5 KiB │ 711.5 KiB │  0 B 


 CLASSES │ old  │ new  │ diff      
─────────┼──────┼──────┼───────────
 classes │  146 │  146 │ 0 (+0 -0) 
 methods │ 1813 │ 1813 │ 0 (+0 -0) 
  fields │  746 │  746 │ 0 (+0 -0) 
```

Reply:
I've read-ed this line in 8.1.1. I'm not sure if that affects the configuration caching in an way.